lib/utils/import/googlekeep/index.ts

```diff
@@ -0,0 +1,101 @@
+import { EventEmitter } from 'events';
+import { endsWith, isEmpty, get, has } from 'lodash';
+import JSZip from 'jszip';
+
+import CoreImporter from '../';
+
+let fs = null;
+const isElectron = has(window, 'process.type');
+if (isElectron) {
+  fs = __non_webpack_require__('fs'); // eslint-disable-line no-undef
+}
+
+class GoogleKeepImporter extends EventEmitter {
+  constructor({ noteBucket, tagBucket, options }) {
+    super();
+    this.noteBucket = noteBucket;
+    this.tagBucket = tagBucket;
+    this.options = options;
+  }
+
+  importNotes = filesArray => {
+    if (isEmpty(filesArray)) {
+      this.emit('status', 'error', 'No file to import.');
+      return;
+    }
+
+    const coreImporter = new CoreImporter({
+      noteBucket: this.noteBucket,
+      tagBucket: this.tagBucket,
+    });
+
+    let importedNoteCount = 0;
+
+    const importJsonString = jsonString => {
+      const importedNote = JSON.parse(jsonString);
+
+      const title = importedNote.title;
+      let textContent = title ? title + '\n\n' : '';
+      textContent += get(importedNote, 'textContent', '');
+
+      if (importedNote.listContent) {
+        // Note has checkboxes
+        textContent += importedNote.listContent
+          .map(item => `- [${item.isChecked ? 'x' : ' '}] ${item.text}`)
+          .join('\n');
+      }
+
+      return coreImporter
+        .importNote(
+          {
+            content: textContent,
+            // note: the exported files don't tell us the creation date...
+            modificationDate: importedNote.userEditedTimestampUsec / 1e6,
+            pinned: importedNote.isPinned,
+            tags: get(importedNote, 'labels', []).map(item => item.name),
+          },
+          { ...this.options, isTrashed: importedNote.isTrashed }
+        )
+        .then(() => {
+          importedNoteCount++;
+          this.emit('status', 'progress', importedNoteCount);
+        });
+    };
+
+    const importZipFile = fileData =>
+      JSZip.loadAsync(fileData).then(zip => {
+        const promises = zip
+          .file(/.*\/Keep\/.*\.json/)
+          .map(zipObj => zipObj.async('string').then(importJsonString));
+        return Promise.all(promises);
+      });
+
+    const promises = filesArray.map(file =>
+      fs.promises
```

Comment:
is this API stable? I thought I might remember seeing something like it would go away or was marked as experimental. would it be safer if we created our own wrapper?

```ts
const readFile = path => new Promise( resolve => setTimeout( () => resolve( fs.readFileSync( path ) ), 0 ) );

const promises = filesArray.map( file => readFile( file.path ) ).then( … ).catch( … );
```

Reply:
Looks like it's stable, at least as of node v12, see [here](https://github.com/nodejs/node/commit/5583d4d73ef750b27c9068ea0f05fae0de9f7766) and the related PR.